### PR TITLE
Update README with general instructions

### PR DESCRIPTION
**Pull request by ethomson**
Provide general instructions for updating the native binaries for use within LibGit2Sharp.

/cc @bording 
/cc @aarnott

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,10 +22,62 @@ currently *experimental*. Would you encounter any issue with it, please open an
...
+
+4. Once merged, the Travis and AppVeyor builds will run.  Once those
+   are done, you can create the nativebinaries nuget package from the
+   build artifacts that were created by the CI systems.
+
+   `download.build.artifacts.and.package.ps1`
+
+   This will emit a nuget package, eg `LibGit2Sharp.NativeBinaries.2.0.291.nupkg`.
+
+5. Upload the package created in step 4 to [nuget.org](https://nuget.org/).
```
Comment (bording) [inline README.md]: Who do we expect the target audience of these instructions to be? Most people won't be able to do this, so while this is the next step, most people reading can't do this.

File: README.md [hunk 2]
```diff
@@ -22,10 +22,62 @@ currently *experimental*. Would you encounter any issue with it, please open an
...
+is a simple process do so:
+
+1. Clone the `LibGit2Sharp.NativeBinaries` repository.  Do so recursively
+   to ensure that the `libgit2` submodule is initialized automatically:
+
+   `git clone --recursive https://github.com/libgit2/libgit2sharp.nativebinaries`
+
+   (If you have already cloned this repository (which seems quite
+   likely since you are reading this file!) then you can simply run
+   `git submodule init` followed by `git submodule update`.)
```
Comment (AArnott) [inline README.md]: `git submodule update --init` works as well and is a single command.

File: README.md (hunk 1 above)
Comment (AArnott) [reply README.md]: We *could* add a "or your private nuget feed" so that it can apply to anyone.

**Comment by bording** [comment]
I just made a substantial revision to the README. I did look at the changes in this PR to see what might be worth incorporating, so I'm going to go ahead and close this.

**Pull request closed without merging**